### PR TITLE
Trainable model

### PR DESCRIPTION
**Pull request by MLnick**
TODO:

- [x] Training test cases

### CODE REVIEW

**Comment by MLnick** [comment]
@kmh4321 
> * The master repo readme should have a small section on training and just point to the training readme. (see: https://github.com/kmh4321/MAX-Image-Segmenter/tree/trainable-patch-1#train-this-model-on-watson-machine-learning)
> 
That section is already in the PR: https://github.com/IBM/MAX-Named-Entity-Tagger/pull/17/files#diff-04c6e90faac2675aa89e2176d2eec7d8R176

File: training/README.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+Open a terminal window, change dir into `$MODEL_REPO_HOME_DIR/training` and install the Python prerequisites. (Model training requires Python 3.6 or above.)
+
+   ```
+   $ cd training/
+
+   $ pip install -r requirements.txt
+    ...
+   ```
+
+### Customize Model Specific Parameters
```
[the remark this answers is not in the document]
Comment (MLnick) [reply training/README.md]: That's a bit blocked on merging that part in the framework right? Perhaps have that as a later update?

File: training/README.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+If you wish to change training hyper-parameters like `epochs`, update the [training command script](https://github.com/IBM/MAX-Named-Entity-Tagger/blob/training/training/training_code/train-max-model.sh#L100).
```
[the remark this answers is not in the document]
Comment (MLnick) [reply training/README.md]: Probably makes sense

File: training/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (MLnick) [reply training/README.md]: For now only `epochs` is adjustable - was planning to do a later update adding more of the hyperparams.

File: Dockerfile [hunk 3]
```diff
@@ -17,19 +17,32 @@
...
+    fi
 
 COPY requirements.txt /workspace
 RUN pip install -r requirements.txt
 
 COPY . /workspace
 
+RUN if [ "$use_pre_trained_model" = "true" ] ; then \
+      # validate downloaded pre-trained model assets
+      md5sum -c md5sums.txt ; \
```
Comment (kmh4321) [inline Dockerfile]: I think we've been using SHA512 sums for the latest version of MAX models.

File: training/data_preparation/training_requirements.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+# TODO  Add packages that the model training code requires when it is executed by WML
```
Comment (kmh4321) [inline training/data_preparation/training_requirements.txt]: Are there some packages missing? Else, we can remove the `TODO`

File: Dockerfile (hunk 3 above)
Comment (MLnick) [reply Dockerfile]: I'll make a separate PR for that, thanks

File: training/data_preparation/training_requirements.txt (hunk 4 above)
Comment (MLnick) [reply training/data_preparation/training_requirements.txt]: Nothing missing as WML already has TF installed which is all that is required.

**Pull request merged**